pkg/reconciler/revision/resources/queue.go

```diff
@@ -372,6 +372,14 @@ func makeQueueContainer(rev *v1.Revision, cfg *config.Config) (*corev1.Container
 		})
 	}
 
+	// Only add this if the value is actually set
```

Comment:
Is there actually any reason not to just set this in all cases - the result in QP will be the same either way since the default value is "" anyway? Maybe we could just drop the if, and then this could be set in the big block with all the other env vars except AutoDetectHTTP2?

Reply:
Just aesthetics, mainly. If we set the envvar in all cases, it adds an empty line to the pod, meaning if you ran `kubectl describe pod` you'd see something like this:

```yaml
- name: METRICS_COLLECTOR_ADDRESS
  value: somethingsomethingsomething
- name: CONCURRENCY_STATE_ENDPOINT
- name: ENABLE_PROFILING
  value: somethingsomethingsomething
```
(creating from memory, so formatting might be off a bit)

It didn't look right to my eye to have an environmental variable created with no value.

Reply:
isn't `METRICS_COLLECTOR_ADDRESS` empty string by default anyway?

Reply:
Appears so (also `TRACING_CONFIG_ZIPKIN_ENDPOINT`).

I don't feel strongly about this, so if it makes more sense to set it always that's fine with me.

Reply:
It's not really for aestethics fwiw. I did this to the HTTP2 switch "back in the day" to avoid churn when doing an upgrade. Frankly, I think we can drop both of these `if` blocks, as we're doing a change to the queue-proxy anyway so the deployment will have to update in any case. One update more doesn't matter.

Reply:
right, I could see the point if we weren't changing QP image anyway but as it is 🤷

Reply:
> Frankly, I think we can drop both of these `if` blocks

Done for freeze proxy; I'll open a separate PR to update the HTTP2 one